### PR TITLE
Corrected index not found at startup

### PR DESCRIPTION
**Pull request by BenEsquis**
Fix the issue when the index doesn't exist at startup for sources

### CODE REVIEW

File: common/auxip-ingestion/src/main/java/com/csgroup/s2pdgs/auxip/ingestion/AuxipFilesIngestion.java [hunk 1]
```diff
@@ -103,5 +113,21 @@ public void ingestExistingAuxFiles() throws Exception {
...
+            .onErrorResume(
+                        IndexNotFoundException.class,
+                        t -> clientManager.monoWithCatalogClient(catalog, client -> client.createIndex(auxIndex))
+                                .map(b -> {
+                                    if (!b) {
+                                        return Mono.error(new IndexNotFoundException());
+                                    }
+                                    return b;                                    
+                                })
+                                .thenReturn(UNIT));
```
Comment (glmxndr) [inline common/auxip-ingestion/src/main/java/com/csgroup/s2pdgs/auxip/ingestion/AuxipFilesIngestion.java:130]: Here the `thenReturn` explicitly throws away the IndexNotFoundException. probably not what you intend to do.

Did you mean something like this?

```java
private Mono<Unit> refreshedIndex() {
        return this.clientManager.monoWithCatalogClient(
            this.catalog,
            client -> client.refreshIndex(this.auxIndex)
            .onErrorResume(
                        IndexNotFoundException.class,
                        t -> client.createIndex(auxIndex)
                                .flatMap(b -> {
                                    if (!b) {
                                        return Mono.error(new IndexNotFoundException());
                                    }
                                    return Mono.just(UNIT); 
                                }));
    }
```

File: common/connect/src/main/java/com/csgroup/s2pdgs/station/StationProductsFinder.java [hunk 2]
```diff
@@ -50,18 +53,28 @@ public StationProductsFinder(
...
+            .onErrorResume(
+                        IndexNotFoundException.class,
+                        t -> clientManager.monoWithCatalogClient(catalog, client -> client.createIndex(catalogIndex))
+                        .map(b -> {
+                            if (!b) {
+                                return Mono.error(new IndexNotFoundException());
+                            }
+                            return b;                                    
+                        })        
+                        .thenReturn(UNIT));
```
Comment (glmxndr) [inline common/connect/src/main/java/com/csgroup/s2pdgs/station/StationProductsFinder.java:77]: Same problem as above.

File: common/connect/src/main/java/com/csgroup/s2pdgs/repository/FTPSRepository.java [hunk 3]
```diff
@@ -44,6 +44,8 @@ public FTPSRepository(FTPSRepositoryConfiguration config) throws IOException, In
         if(config.getFtpServerCertificateStorePath() != null){
             client.setTrustManager(TrustManagerUtils.getValidateServerCertificateTrustManager());
             
+            log.debug("Using certificate : "+config.getFtpServerCertificateStorePath());
```
Comment (glmxndr) [inline common/connect/src/main/java/com/csgroup/s2pdgs/repository/FTPSRepository.java:47]: Avoid concatenation in log method (use `{}` instead).

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java [hunk 4]
```diff
@@ -53,15 +53,17 @@ public StationPoller(
...
         return products
-            .doOnNext(p -> log.trace("Treating: {}", p.getName()))
+            .doOnNext(p -> log.debug("Treating: {}", p.getName()))
             .map(x -> createMessage(x))
             .doOnTerminate(repo::close);
     }
 
     private Mono<Repository<?>> connectedRepo() {
         return Mono.fromCallable(() -> {
+            log.info("Connecting to repository with configuration : "+configuration.getClass().getCanonicalName());
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java:63]: Avoid concatenation in log methods.

**Review by glmxndr — CHANGES_REQUESTED** [review]

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java [hunk 5]
```diff
@@ -43,7 +43,7 @@ public StationPoller(
     public Flux<Message<StationOutput>> get() {
         Mono<Repository<?>> repository = connectedRepo();
         return repository
-            .doOnNext(r -> log.trace("connectedRepo: {}", r))
+            .doOnNext(r -> log.debug("connectedRepo: {}", r))
```
Comment (ilaborie) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java:46]: Maybe: use of one `Mono#log` function ?

https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#log--

Comment (glmxndr) [reply applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java:46]: The Flux/Mono `log()` methods don't allow formatting of the message, and require to provide which types of messages you are interested in (by default they provide subscriptions, requests and complete signals). To me they serve a different purpose than what is done here: they allow to inspect what happens in the reactor layer.

Comment (ilaborie) [reply applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationPoller.java:46]: OK

File: common/auxip-ingestion/src/main/java/com/csgroup/s2pdgs/auxip/ingestion/AuxipFilesIngestion.java (hunk 1 above)
Comment (BenEsquis) [reply common/auxip-ingestion/src/main/java/com/csgroup/s2pdgs/auxip/ingestion/AuxipFilesIngestion.java:130]: Il me semblait que le thenReturn ne renvoyait UNIT que lorsque c'était en success sinon il renvois le Mono<Error> ?
Mais en effet t'a version semble plus explicite.

**Review by glmxndr — APPROVED** [review]

**Pull request merged**